chore: do npm publish on release event (#54)
hCaptcha/react-native-hcaptcha: MAINTAINER.md

### Release steps

> 1. Decived about next version:

Reviewer: ```suggestion
1. Pick the right version number:
```

**Note:** we use [SemVer format](https://semver.org/):

MAJOR: breaking change.
MINOR: new feature(s), backwards compatible.
PATCH: bugfix only.

2. Prepare a PR:

- bump [`package.json's`](./package.json) version
- run `npm i` to update `package-lock.json`
- commit `package.json` and `package-lock.json`

> - do tag git tag vM.M.P

Reviewer: ```suggestion
- set the tag on master, matching your version: git tag `vM.M.P`
```

> - open the PR to review

Reviewer: ```suggestion
- open the PR for review
```

> - once the PR approved and merged create a new release https://github.com/hCaptcha/react-native-hcaptcha/releases

Reviewer: ```suggestion
- once the PR is approved and merged to master:
 - set the tag on master matching your version: git tag `vM.M.P`
 - draft a new release https://github.com/hCaptcha/react-native-hcaptcha/releases
```

> - wait for CI to release new version to https://www.npmjs.com/package/@hcaptcha/react-native-hcaptcha?activeTab=versions

Reviewer: ```suggestion
- once the release is created, CI will release the new version to https://www.npmjs.com/package/@hcaptcha/react-native-hcaptcha?activeTab=versions
```

- bump [`package.json's`](./package.json) version
- run `npm i` to update `package-lock.json`
- commit `package.json` and `package-lock.json`
- open the PR for review
- once the PR is approved and merged to master:

>  - set the tag on master matching your version: git tag `vM.M.P`

Reviewer: tag should be on master. releases should get signoff anyway, so it's fine to ask.
